packages/rpc-methods/src/permitted/middleware.ts

```diff
@@ -0,0 +1,53 @@
+import { logError } from '@metamask/snaps-utils';
+import { ethErrors } from 'eth-rpc-errors';
+import { JsonRpcMiddleware, JsonRpcRequest } from 'json-rpc-engine';
+
+import { handlers } from '.';
```

Comment:
I think this creates a circular dependency. `middleware.ts` imports from `index.ts` and `index.ts` exports from `middleware.ts`.